### PR TITLE
CSV Save Feature

### LINKED ISSUE #40: Implement SAVE function to save neighborhood data into a csv file
Create a function in csv.js called save(). The save() function receives a string filepath as a parameter. The save() function should grab the NeighborhoodList[] data from operation.js, and save it to the specified filepath.
- If the file doesn't exist at the filepath, save() should create a new file according to the filepath.
- If there is a file at the filepath, save() should overwrite the contents of that file with the NeighborhoodList[] data.
- save() should save files into the same format that the load() needs to properly load data.

### PR DESCRIPTION
**Pull request by raymondwzeng**
## Major Changes
Added a new function to `csv.js`, aptly named save. This saves a specific set of data to a specific path on disk. This should only be called from the Operations layer.

I also added 3 unit tests that messes with the data and ensures that the data is readable afterwards. There's a non-zero chance that the tests won't run as an entire suite due to the messing around with data, but feel free to run it with mocha on its own.

Fixes #40

### CODE REVIEW

**Review by tgustafson2 — APPROVED** [review]
Looks good to me

**Review by dukeph0 — APPROVED** [review]
I approve

**Review by AaronGeesink — CHANGES_REQUESTED** [review]
On second glance, It seems that some of the changes you made broke some of the unit tests. Please figure out why they are breaking, and fix them before you go ahead and merge into main. The broken test cases are all in the Operations layer. Maybe you accidentally saved to the California_Houses.csv file when making the save function? I will admit that the Operation Layer unit tests are really finnicky and could benefit from a second opinion.

**Comment by raymondwzeng** [comment]
Lots of refactoring. Things should be alright now.

image

**Review by AaronGeesink — APPROVED** [review]
unit tests now pass consistently no matter how many times we call `npm test` AND we finally have unit tests for all `api/cards`  endpoints. Good work, let's merge.

File: server/test/apiTest.js
Comment (AaronGeesink) [inline server/test/apiTest.js:87]: It's possible this unit test (and similar ones that have a hardcoded length in them) will potentially have to be changed in the future if we end up saving to the original csv file. However, for now this seems like an alright approach.

**Pull request merged**
